third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -871,6 +1071,68 @@ static void SendCDPMessage(const v8::FunctionCallbackInfo<v8::Value>& args) {
   gInspectorSession->dispatchProtocolMessage(messageView);
 }
 
+static std::string GetRecordingDirectory() {
+  const char* recordingDir = getenv("RECORD_REPLAY_DIRECTORY");
+  if (recordingDir) {
+    return recordingDir;
+  }
+  const char* homeDir = getenv("HOME");
+  if (!homeDir) {
+    homeDir = getenv("USERPROFILE");
+  }
+  return std::string(homeDir) + DirectorySeparator + std::string(".replay");
+}
+
+static void GetRecordingId(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+  args.GetReturnValue().Set(ToV8String(isolate, recordreplay::GetRecordingId()));
+}
+
+static void SHA256DigestHex(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  CHECK(args.Length() == 1 && args[0]->IsString() &&
+      "must be called with a single string");
+  v8::Isolate* isolate = args.GetIsolate();
+  v8::String::Utf8Value content(isolate, args[0]);
+
+  std::unique_ptr<crypto::SecureHash> hasher =
+    crypto::SecureHash::Create(crypto::SecureHash::SHA256);
+  hasher->Update(*content, strlen(*content));
```

Comment:
We should use `content.length()` instead of `strlen` since utf8 strings can contain null bytes.

Reply:
TIL. Thanks!